homeassistant/components/vorwerk/vacuum.py

```diff
@@ -0,0 +1,348 @@
+        _LOGGER.debug("self._state=%s", self._state)
+        if "alert" in self._state:
+            robot_alert = ALERTS.get(self._state["alert"])
+        else:
+            robot_alert = None
+        if self._state["state"] == 1:
+            if self._state["details"]["isCharging"]:
+                self._clean_state = STATE_DOCKED
+                self._status_state = "Charging"
+            elif (
+                self._state["details"]["isDocked"]
+                and not self._state["details"]["isCharging"]
+            ):
+                self._clean_state = STATE_DOCKED
+                self._status_state = "Docked"
+            else:
+                self._clean_state = STATE_IDLE
+                self._status_state = "Stopped"
+
+            if robot_alert is not None:
+                self._status_state = robot_alert
+        elif self._state["state"] == 2:
+            if robot_alert is None:
+                self._clean_state = STATE_CLEANING
+                self._status_state = (
+                    f"{MODE.get(self._state['cleaning']['mode'])} "
+                    f"{ACTION.get(self._state['action'])}"
+                )
+                if (
+                    "boundary" in self._state["cleaning"]
+                    and "name" in self._state["cleaning"]["boundary"]
+                ):
+                    self._status_state += (
+                        f" {self._state['cleaning']['boundary']['name']}"
+                    )
+            else:
+                self._status_state = robot_alert
+        elif self._state["state"] == 3:
+            self._clean_state = STATE_PAUSED
+            self._status_state = "Paused"
+        elif self._state["state"] == 4:
+            self._clean_state = STATE_ERROR
+            self._status_state = ERRORS.get(self._state["error"])
+
+        self._battery_level = self._state["details"]["charge"]
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def supported_features(self):
+        """Flag vacuum cleaner robot features that are supported."""
+        return SUPPORT_VORWERK
+
+    @property
+    def battery_level(self):
+        """Return the battery level of the vacuum cleaner."""
+        return self._battery_level
+
+    @property
+    def available(self):
+        """Return if the robot is available."""
+        return self._available
+
+    @property
+    def icon(self):
+        """Return specific icon."""
+        return "mdi:robot-vacuum-variant"
+
+    @property
+    def state(self):
+        """Return the status of the vacuum cleaner."""
+        return self._clean_state
+
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return self._robot_serial
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the vacuum cleaner."""
+        data = {}
+
+        if self._status_state is not None:
```

Comment:
except if False is a valid value, you can just do `if self._status_state:`

Reply:
I know. I just copied vacuum.py from the neato integration vacuum.py. And I don't know if False is a valid value. For now I would like to keep neato and vorwork as near as possilbe.

Reply:
I don't know if false is a valid value. The code is based on the neato component. Vorwerk robots are branded neato robots, with a different authentication.

Reply:
ok